release/models/network-instance/openconfig-network-instance-l2.yang

```diff
@@ -376,4 +383,308 @@ submodule openconfig-network-instance-l2 {
+              }
+              description "Leafref of MAC-IP address object";
+            }
+
+            leaf vlan {
+              type leafref {
+                path "../state/vlan";
+              }
+              description
+                "Leafref VLAN on which the MAC-IP address is present";
+            }
+
+            leaf host-ip {
+              type leafref {
+                path "../state/host-ip";
+              }
+                description "Leafref host-ip for the MAC-IP table entry";
+            }
+
+            container state {
+              config false;
+              description
+                "Operational state data for L2RIB MAC-IP table object entry";
+              uses l2ni-l2rib-mac-ip-table-state;
+            }
+
+            container producers {
+              description "Source producers for each MAC-IP Table entry";
+              list producer {
+                key "producer";
+                description
+                  "List of producers for each MAC-IP table entry";
+
+                leaf producer {
+                  type leafref {
+                    path "../state/producer";
+                  }
+                  description
+                    "Reference to producer list key";
+                }
+
+                container state {
+                  config false;
+                  description "State container for L2RIB MAC Table Entries";
+                  uses l2ni-l2rib-common-producer-state;
+                }
+              }
+            }
+
+            uses l2ni-l2rib-common-next-hop-state;
+          }
+        }
+      }
+    }
+  }
+
+  grouping l2ni-l2rib-mac-table-state {
+    description "L2RIB MAC Table Oper State Grouping";
+    uses l2ni-l2rib-common-state;
+  }
+
+  grouping l2ni-l2rib-mac-ip-table-state {
+    description "L2RIB Mac-IP Table Oper State Grouping";
+    uses l2ni-l2rib-common-state;
+
+    leaf host-ip {
+      type inet:ip-address;
+      description "Host IP address for the L2RIB MAC-IP entry";
+    }
+
+    leaf l3-vni {
+      type oc-evpn-types:evi-id;
+      description "Layer3 VNI used for inter-subnet routing";
+    }
+  }
+
+  grouping l2ni-l2rib-common-state {
+    description "L2RIB Common Property Oper Data Grouping";
+
+    leaf mac-address {
+        type yang:mac-address;
+        description "MAC address of the L2RIB MAC or MAC-IP entry";
+    }
+    leaf vlan {
+      type leafref {
+        path "../../../../../../../vlans/vlan/config/vlan-id";
+      }
+      description
+        "VLAN on which the MAC or MAC-IP address is present.";
+    }
+    leaf evi {
+      type oc-evpn-types:evi-id;
+      description "EVPN Instance Identifier for the MAC or MAC-IP";
+    }
+    leaf l2-vni {
+      type oc-evpn-types:evi-id;
+      description "Layer2 VNI segment mapped to given vlan-id";
+    }
+  }
+
+  grouping l2ni-l2rib-common-producer-state {
+    description "L2RIB Common Producer Attributes Oper Data Grouping";
```

Comment:
Can you expand the description, to facilitate the 'producer' concept understanding.